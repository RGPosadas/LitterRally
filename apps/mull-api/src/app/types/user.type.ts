import { Field, ID, ObjectType } from '@nestjs/graphql';

@ObjectType('User')
export class UserType {
  @Field(/* istanbul ignore next */ () => ID)
  id: number;

  @Field()
  password: string;

  @Field()
  email: string;

  @Field()
  timezone: string;

  @Field()
  name: string;

  @Field()
  age: string;

  @Field()
  description: string;

  @Field(/* istanbul ignore next */ () => [UserType])
  friends: UserType[];

  // TODO: Implement modules, services, types for Entities below

  //   @Field(() => [EventType])
  //   events: EventType[];

  //   @Field(() => [PostReactionType])
  //   postReactions: PostReactionType[];

  //   @Field(() => MediaType)
  //   avatar: MediaType;
}
