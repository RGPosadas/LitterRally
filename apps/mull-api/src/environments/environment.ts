require('dotenv').config();

export const environment = {
  production: false,

  db: {
    host: process.env.DB_HOST,
    username: process.env.DB_USERNAME,
    password: process.env.DB_PASSWORD,
  },
  auth: {
    google: {
      clientId: process.env.GOOGLE_CLIENT_ID,
      clientSecret: process.env.GOOGLE_SECRET,
    },
    facebook: {
      clientId: process.env.FACEBOOK_CLIENT_ID,
      clientSecret: process.env.FACEBOOK_SECRET,
    },
    twitter: {
      consumerId: process.env.TWITTER_CONSUMER_ID,
      consumerSecret: process.env.TWITTER_CONSUMER_SECRET,
    },
  },
  session: {
    secret: process.env.SESSION_SECRET,
  },
  jwt: {
    accessSecret: process.env.ACCESS_TOKEN_SECRET,
    refreshSecret: process.env.REFRESH_TOKEN_SECRET,
  },
  client: {
    baseUrl: process.env.CLIENT_BASE_URL,
  },
};
