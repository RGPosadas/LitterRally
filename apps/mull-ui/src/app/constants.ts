export const ROUTES = {
  HOME: '/home',
  MAP: '/map',
  NEW_EVENT: '/new-event',
  TOOLS: '/tools',
  PROFILE: '/profile',
};
