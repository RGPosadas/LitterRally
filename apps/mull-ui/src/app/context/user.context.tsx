import React from 'react';

const UserContext = React.createContext<{
  userId: number;
  setUserId: React.Dispatch<React.SetStateAction<number>>;
  accessToken: string;
  setAccessToken: React.Dispatch<React.SetStateAction<string>>;
}>(null);

export const UserProvider = UserContext.Provider;
export const UserConsumer = UserContext.Consumer;

export default UserContext;
