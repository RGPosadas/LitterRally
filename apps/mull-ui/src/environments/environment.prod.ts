export const environment = {
  production: true,
  backendUrl: '',
  backendWsUrl: 'wss://www.mullapp.com',
};
