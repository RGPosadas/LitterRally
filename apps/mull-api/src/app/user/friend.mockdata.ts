import { RegistrationMethod } from '@mull/types';
import { mockAllDirectMessageChannels } from '../channel/channel.mockdata';
import { Friend } from '../entities';
import { mockAllPosts } from '../post/post.mockdata';

// Friends of mockAllUsers[0]
export const mockAllFriends: Friend[] = [
  {
    id: 7,
    password: 'password',
    email: 'me@me.com',
    timezone: '',
    name: 'Cristian',
    dob: new Date(),
    description: "There's a first for everything",
    friends: [],
    tokenVersion: 0,
    registrationMethod: RegistrationMethod.LOCAL,
    avatar: null,
    latestPost: mockAllPosts[3],
    directMessageChannel: mockAllDirectMessageChannels[0],
  },
  {
    id: 12,
    password: 'abc123',
    email: 'tim@green.co',
    timezone: '',
    name: 'Tim',
    dob: new Date(),
    description: '',
    friends: [],
    tokenVersion: 0,
    registrationMethod: RegistrationMethod.LOCAL,
    avatar: null,
    latestPost: null,
    directMessageChannel: null,
  },
];
