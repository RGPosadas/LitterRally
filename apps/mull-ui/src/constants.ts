import { EventRestriction, IEvent } from '@mull/types';

/**
 * enum of the current routes for the application
 */
export enum ROUTES {
  LOGIN = '/login',
  HOME = '/home',
  MAP = '/map',
  CREATE_EVENT = '/create-event',
  TOOLS = '/tools',
  MESSAGES = '/messages',
  PROFILE = '/profile',
}

export const DAY_IN_MILLISECONDS = 86400000;

export const dummyEvent: IEvent = {
  id: 1,
  title: 'Test title',
  description:
    'Lorem ipsum dolor sit amet, regione invidunt democritum vim in, movet antiopam gubergren ne per. Est nibh magna scribentur ea, vel te munere deseruisse disputationi, eros meis ludus ne sea. Odio prompta legendos in sea, ut mei scripta labores theophrastus, id molestie probatus periculis mea. Noluisse invenire splendide sed ne, at erat quando laudem nec, possim apeirian vix at. At eam animal efficiendi interpretaris, eirmod offendit adversarium per et, summo qualisque efficiendi in has. Vel detraxit accusata ea.',
  startDate: new Date(2000, 0, 1, 0, 0, 0, 0),
  endDate: new Date(2000, 0, 2, 0, 0, 0, 0),
  location: {
    id: 1,
    point: '1260 Remembrance Road Montreal, Qc',
  },
  restriction: EventRestriction.NONE,
  participants: [
    {
      name: 'User 1',
    },
    {
      name: 'User 2',
    },
  ],
  host: {
    name: 'Test user',
  },
};
