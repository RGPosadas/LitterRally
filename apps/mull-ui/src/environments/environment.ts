// This file can be replaced during build by using the `fileReplacements` array.
// When building for production, this file is replaced with `environment.prod.ts`.
import { API_HOSTNAME } from '@mull/types';

export const environment = {
  production: false,
  backendUrl: 'http://' + API_HOSTNAME + ':3333',
  backendWsUrl: 'ws://' + API_HOSTNAME + ':3333',
};
