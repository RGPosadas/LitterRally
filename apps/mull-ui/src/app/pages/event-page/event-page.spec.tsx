import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { ISerializedEvent } from '@mull/types';
import { act, render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { dummyEvent } from '../../../constants';
import { environment } from '../../../environments/environment';
import {
  EventPageDocument,
  EventPageQuery,
  UserDocument,
  UserQuery,
} from '../../../generated/graphql';
import { UserProvider } from '../../context/user.context';
import EventPage from './event-page';

describe('EventPage', () => {
  const renderHelper = (event: ISerializedEvent, mocks: MockedResponse[]) => {
    return (
      <UserProvider value={{ userId: 1, setUserId: jest.fn() }}>
        <MockedProvider mocks={mocks} addTypename={false}>
          <MemoryRouter initialEntries={['events/1']}>
            <Route path="events/:id">
              <EventPage reviewEvent={event} prevPage="" eventImageURL="" />
            </Route>
          </MemoryRouter>
        </MockedProvider>
      </UserProvider>
    );
  };

  it('should render successfully', () => {
    const { baseElement } = render(renderHelper(dummyEvent, null));
    expect(baseElement).toBeTruthy();
  });

  it('should show queried user information', async () => {
    const mocks: MockedResponse[] = [
      {
        request: {
          query: UserDocument,
          variables: { userId: 1 },
        },
        result: {
          data: {
            user: {
              id: 1,
              name: 'Test user',
            },
          } as UserQuery,
        },
      },
    ];

    const utils = render(renderHelper(dummyEvent, mocks));
    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 0));
    });

    expect(utils.getByTestId('event-page-host').textContent).toEqual('Test user');
  });

  it('should show queried event information', async () => {
    const mocks: MockedResponse[] = [
      {
        request: {
          query: EventPageDocument,
          variables: { eventId: 1 },
        },
        result: {
          data: {
            event: {
              id: 1,
              title: 'Test title',
              description: 'Test Description',
              startDate: '2020-12-12T03:00:00.000Z',
              endDate: '2020-12-12T03:00:00.000Z',
              host: {
                id: 1,
                name: 'Test user 2',
              },
              restriction: 1,
              image: {
                id: 1,
                mediaType: 'jpeg',
              },
              location: {
                title: 'Test Location',
              },
              __typename: 'Event',
            },
            isParticipant: true,
          } as EventPageQuery,
        },
      },
    ];

    const utils = render(renderHelper(null, mocks));
    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 0));
    });

    expect(utils.getByTestId('event-page-title').textContent).toEqual('Test title');
    expect(utils.getByTestId('event-page-host').textContent).toEqual('Test user 2');
    expect(utils.getByTestId('event-page-location').textContent).toEqual('Test Location');
    expect(utils.getByTestId('event-page-location').textContent).toEqual('Test Location');
    expect(utils.getByTestId('event-page-image').getAttribute('src')).toEqual(
      `${environment.backendUrl}/api/media/1`
    );
  });
});
