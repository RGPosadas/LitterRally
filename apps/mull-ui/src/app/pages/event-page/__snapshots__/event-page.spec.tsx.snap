// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventPage should match snapshot 1`] = `
<div
  className="page-container no-padding event-page-container"
>
  <div
    className="event-page-header"
  >
    <button
      className="mull-back-button undefined"
      data-testid="mull-back-button"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-left fa-w-10 mull-back-button-icon"
        data-icon="chevron-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <div
        className="mull-back-button-text"
      >
        
      </div>
    </button>
    <div
      className="title"
    >
      Test title
    </div>
    <img
      alt="Event Page"
      className="event-image"
      src=""
    />
    <div
      className="event-datetime"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-clock fa-w-16 clock-icon event-page-icon color-grey"
        data-icon="clock"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <div
        className="event-datetime-string"
        data-testid="start-date-div"
      >
        <div>
          Jan 1
        </div>
        <div>
          12:00 AM
        </div>
      </div>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-caret-right fa-w-6 event-page-icon color-grey"
        data-icon="caret-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 192 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <div
        className="event-datetime-string"
        data-testid="end-date-div"
      >
        <div>
          Jan 2
        </div>
        <div>
          12:00 AM
        </div>
      </div>
    </div>
  </div>
  <div
    className="event-page-info"
  >
    <div
      className="event-page-info-container "
    >
      <div
        className="info-row"
      >
        <img
          alt="Event"
          className="event-page-icon"
          src="https://i.pinimg.com/originals/0c/3b/3a/0c3b3adb1a7530892e55ef36d3be6cb8.png"
        />
        <p
          className="row-text"
          data-testid="event-host"
        >
          Placeholder Host
        </p>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-comments fa-w-18 event-page-icon color-green"
          data-icon="comments"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 192c0-88.4-93.1-160-208-160S0 103.6 0 192c0 34.3 14.1 65.9 38 92-13.4 30.2-35.5 54.2-35.8 54.5-2.2 2.3-2.8 5.7-1.5 8.7S4.8 352 8 352c36.6 0 66.9-12.3 88.7-25 32.2 15.7 70.3 25 111.3 25 114.9 0 208-71.6 208-160zm122 220c23.9-26 38-57.7 38-92 0-66.9-53.5-124.2-129.3-148.1.9 6.6 1.3 13.3 1.3 20.1 0 105.9-107.7 192-240 192-10.8 0-21.3-.8-31.7-1.9C207.8 439.6 281.8 480 368 480c41 0 79.1-9.2 111.3-25 21.8 12.7 52.1 25 88.7 25 3.2 0 6.1-1.9 7.3-4.8 1.3-2.9.7-6.3-1.5-8.7-.3-.3-22.4-24.2-35.8-54.5z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="info-row"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-map-marker-alt fa-w-12 event-page-icon color-grey"
          data-icon="map-marker-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <p
          className="row-text"
          data-testid="event-location"
        >
          18 Mull Ave, Pointe-Claire, Montreal
        </p>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-map fa-w-18 event-page-icon color-green"
          data-icon="map"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 117.66v346.32c0 11.32 11.43 19.06 21.94 14.86L160 416V32L20.12 87.95A32.006 32.006 0 0 0 0 117.66zM192 416l192 64V96L192 32v384zM554.06 33.16L416 96v384l139.88-55.95A31.996 31.996 0 0 0 576 394.34V48.02c0-11.32-11.43-19.06-21.94-14.86z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="info-row"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-align-left fa-w-14 event-page-icon color-grey"
          data-icon="align-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.83 352h262.34A12.82 12.82 0 0 0 288 339.17v-38.34A12.82 12.82 0 0 0 275.17 288H12.83A12.82 12.82 0 0 0 0 300.83v38.34A12.82 12.82 0 0 0 12.83 352zm0-256h262.34A12.82 12.82 0 0 0 288 83.17V44.83A12.82 12.82 0 0 0 275.17 32H12.83A12.82 12.82 0 0 0 0 44.83v38.34A12.82 12.82 0 0 0 12.83 96zM432 160H16a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0 256H16a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <div
          className="row-text"
          data-testid="expandable-text-div"
        >
          Lorem ipsum dolor sit amet, regione invidunt democritum vim in, movet antiopam gubergren ne per. Est nibh magna scribentur ea, vel te munere deseruiss
          ... 
          <button
            className="expandable-span-button"
            data-testid="expandable-text-button"
            onClick={[Function]}
          >
            more
          </button>
        </div>
      </div>
      <div
        className="info-row"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-lock fa-w-14 event-page-icon color-grey"
          data-icon="lock"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <p
          className="row-text"
          data-testid="event-restriction"
        >
          Everyone
        </p>
      </div>
      <div
        className="info-row"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-user-friends fa-w-20 event-page-icon color-grey"
          data-icon="user-friends"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M192 256c61.9 0 112-50.1 112-112S253.9 32 192 32 80 82.1 80 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C51.6 288 0 339.6 0 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zM480 256c53 0 96-43 96-96s-43-96-96-96-96 43-96 96 43 96 96 96zm48 32h-3.8c-13.9 4.8-28.6 8-44.2 8s-30.3-3.2-44.2-8H432c-20.4 0-39.2 5.9-55.7 15.4 24.4 26.3 39.7 61.2 39.7 99.8v38.4c0 2.2-.5 4.3-.6 6.4H592c26.5 0 48-21.5 48-48 0-61.9-50.1-112-112-112z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <p
          className="row-text"
        >
          User 1, User 2
        </p>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-user-plus fa-w-20 event-page-icon color-green"
          data-icon="user-plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M624 208h-64v-64c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v64h-64c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h64v64c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-64h64c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm-400 48c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <button
        className="mull-button event-page-button"
        data-testid="mull-button"
        onClick={[Function]}
        type="submit"
      >
        Done
      </button>
    </div>
  </div>
</div>
`;
